src/App.js

```diff
@@ -1,16 +1,32 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [messages, setMessages] = useState(chatMessages);
+  const [totalHeartCount, setHeartCount] = useState(0);
+  const toggleLike = (messageID) => {
+    messages[messageID].liked = !messages[messageID].liked;
+    setMessages(messages);
+    setHeartCount(() =>
+      messages[messageID].liked ? totalHeartCount + 1 : totalHeartCount - 1
+    );
+    return messages[messageID].liked ? '❤️' : '🤍';
+  };
```

Comment:
This works with the data we have, but consider what would happen if the `id` was more complex like a hashed string (which is common) `ef5n2Jk3`. Relying on the assumption that the `id` will fit the array index won't always work.

So what might be a safer solution? How could we iterate through the messages and find the appropriate `id` value?

We also don't need to return a value. This function is strictly for updating state, so we need to come up with another way to display the correct heart emoji through props or state.

Reply:
Remember that when you update an array (adding, removing, and updating values inside) in a piece of state, like `messages`, you must first make a copy first, update the copy, then set state to the new copy.

This will ensure React can _hear_ that state changed. This will allow the last test in wave 3 to pass.